### ISSUE TITLE
Wrong gas estimation for `call`

### ISSUE THREAD

**Issue by ngotchac** [paritytech/parity#3749]
The gas estimation for a method using the `call` function might be wrong.
Take these contract as example:

```
contract B {
  uint public n;
  
  function callSetN(address _contract, uint _n) {
    _contract.call(bytes4(sha3("setN(uint256)")), _n); // E's storage is set, D is not modified 
  }
}

contract A {
  uint public n;
  
  function setN(uint _n) {
    n = _n;
  }
}
```

A call to the `callSetN` method is estimated with 52,976 gas, but when looking at the trace, the transaction runs out of gas : https://testnet.etherscan.io/vmtrace?txhash=0xc2a1a4bb3760e730f7567e7f290c60279356cfa717516da45aff48edec3221bd&type=parity

**Comment by tomusdrw** [comment]
I think `eth_estimateGas` doesn't include base transaction gas cost `21k` and cost of `input` data (4 for each `0` byte and 68 for `non-0`) it only calculates cost of execution.
For this method `estimateGas` returns around `44k` of gas + `21k` of base gas + `~4k` for input the minimal amount should be `69k`.

I think the GUI should add this `~25k` for each transaction before sending.

**Comment by tomusdrw** [comment]
Just realized that it might be related to changes in recent hard fork. Now call provides up to `63/64` gas instead of everything - this would explain why it works with high gas, and then after using exactly the value provided it fails.

Probably we could track calls and increase the gas accordingly.

**Comment by 5chdn** [comment]
This issue is labelled with _bug_: The client fails to follow expected behavior; and is inactive for several months. It's neither assigned nor linked to a milestone.

Please, decide on a deadline and assign this ticket to a developer within 7 days or close it if fixed otherwise.

**Comment by arkpar** [comment]
Fixed by #4100
